src/processes.c

```diff
@@ -25,2153 +25,3026 @@
+
+  return 0;
+} /* int *ps_read_status */
+
+static int ps_read_io(procstat_entry_t *ps) {
+  FILE *fh;
+  char buffer[1024];
+  char filename[64];
+
+  char *fields[8];
+  int numfields;
+
+  snprintf(filename, sizeof(filename), "/proc/%li/io", ps->id);
+  if ((fh = fopen(filename, "r")) == NULL) {
+    DEBUG("ps_read_io: Failed to open file `%s'", filename);
+    return -1;
+  }
+
+  while (fgets(buffer, sizeof(buffer), fh) != NULL) {
+    derive_t *val = NULL;
+    long long tmp;
+    char *endptr;
+
+    if (strncasecmp(buffer, "rchar:", 6) == 0)
+      val = &(ps->gauges.io_rchar);
+    else if (strncasecmp(buffer, "wchar:", 6) == 0)
+      val = &(ps->gauges.io_wchar);
+    else if (strncasecmp(buffer, "syscr:", 6) == 0)
+      val = &(ps->gauges.io_syscr);
+    else if (strncasecmp(buffer, "syscw:", 6) == 0)
+      val = &(ps->gauges.io_syscw);
+    else if (strncasecmp(buffer, "read_bytes:", 11) == 0)
+      val = &(ps->gauges.io_diskr);
+    else if (strncasecmp(buffer, "write_bytes:", 12) == 0)
+      val = &(ps->gauges.io_diskw);
+    else
+      continue;
+
+    numfields = strsplit(buffer, fields, STATIC_ARRAY_SIZE(fields));
+
+    if (numfields < 2)
+      continue;
+
+    errno = 0;
+    endptr = NULL;
+    tmp = strtoll(fields[1], &endptr, /* base = */ 10);
+    if ((errno != 0) || (endptr == fields[1]))
+      *val = -1;
+    else
+      *val = (derive_t)tmp;
+  } /* while (fgets) */
+
+  if (fclose(fh)) {
+    WARNING("processes: fclose: %s", STRERRNO);
+  }
+  return 0;
+} /* int ps_read_io (...) */
+
+static int ps_count_maps(pid_t pid) {
+  FILE *fh;
+  char buffer[1024];
+  char filename[64];
+  int count = 0;
+
+  snprintf(filename, sizeof(filename), "/proc/%d/maps", pid);
+  if ((fh = fopen(filename, "r")) == NULL) {
+    DEBUG("ps_count_maps: Failed to open file `%s'", filename);
+    return -1;
+  }
+
+  while (fgets(buffer, sizeof(buffer), fh) != NULL) {
+    if (strchr(buffer, '\n')) {
+      count++;
+    }
+  } /* while (fgets) */
+
+  if (fclose(fh)) {
+    WARNING("processes: fclose: %s", STRERRNO);
+  }
+  return count;
+} /* int ps_count_maps (...) */
+
+static int ps_count_fd(int pid) {
+  char dirname[64];
+  DIR *dh;
+  struct dirent *ent;
+  int count = 0;
+
+  snprintf(dirname, sizeof(dirname), "/proc/%i/fd", pid);
+
+  if ((dh = opendir(dirname)) == NULL) {
+    DEBUG("Failed to open directory `%s'", dirname);
+    return -1;
+  }
+  while ((ent = readdir(dh)) != NULL) {
+    if (!isdigit((int)ent->d_name[0]))
+      continue;
+    else
+      count++;
+  }
+  closedir(dh);
+
+  return (count >= 1) ? count : 1;
+} /* int ps_count_fd (pid) */
+
+#if HAVE_LIBTASKSTATS
+static int ps_delay(procstat_entry_t *ps) {
+  if (taskstats_handle == NULL) {
+    return ENOTCONN;
+  }
+
+  int status = ts_delay_by_tgid(taskstats_handle, (uint32_t)ps->id, &ps->gauges.delay);
+  if (status == EPERM) {
+    static c_complain_t c;
+#if defined(HAVE_SYS_CAPABILITY_H) && defined(CAP_NET_ADMIN)
+    if (check_capabil
```

Comment:
There seems to be inconsistent identation in this file (the two lines above are one instance), perhaps due to mixed usage of spaces and tabs. Can you please unify it following the upstream convention?